### PR TITLE
refactor: with clang-tidy linter (resolves #9)

### LINKED ISSUE #9: Refactor code using clang-tidy linter
Refactor the current C++ xarm-commander implementation by using the clang-tidy linter.

### PR DESCRIPTION
**Pull request by jo-bru**
Refactored code using the clang-tidy linter.

The following warnings were resolved:
* llvm-include-order
* google-readability-todo
* readability-magic-numbers
* cppcoreguidelines-owning-memory (`arm = new XArmAPI(port);`)
* readability-qualified-auto
* modernize-loop-convert

The following problems are not yet resolved:
* _do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast insteadclang-tidy(cppcoreguidelines-pro-bounds-array-to-pointer-decay)_ **=>depending on SDK functions, therefore not resolvable (I guess)**
* _an exception may be thrown in function 'main' which should not throw exceptionsclang-tidy(bugprone-exception-escape)_
**=> see issue #10**

I'm not sure if the magic number resolution is correct: not sure if `constexpr` is the right way and where to define them (use namespace or separate header, I suppose not as global constants..)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,3 +42,24 @@ xarm-commander -h
 ```
 > :bulb: Can be also used for subcommands: `xarm-commander get_version -h`  
 
+## Configuration Parameters
+### Formatter:
+Formatting engine: `clangFormat`  
+Formatting style: `Visual Studio`
+
+### Linter:
```
Comment (iomz) [inline README.md:50]: What is the actual command you are running on the console? (I.e., how do I reproduce your clang-tidy execution?

**Review by iomz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jo-bru) [reply README.md:50]: I use an vscode extension, so it's defined in my settings..
But I guess you could use it by calling: `clang-tidy src/xarm-commander.cpp -checks="LIST_ALL_THE_CHECKS_HERE" -- -Iinclude`
or creating an .clang-tidy file..but I haven't used the file option yet..

Comment (iomz) [reply README.md:50]: Can you please put the concrete command here? Not everyone is using VS Code...
Also for the formatting. I want the codebase to be editor-agnostic.

Comment (jo-bru) [reply README.md:50]: Ah okay, sure!
I'll probably switch to Google then, or do you have a preference on formatting?

**Review by iomz — APPROVED** [review]
Great, LGTM. Thank you!

**Comment by jo-bru** [comment]
I will probably change the checks and formatting a bit over time, but since they are stored in the configuration files, that won't be a problem.. :)

**Pull request merged**